security/cmd/istio_ca/main.go

```diff
@@ -97,6 +108,113 @@ var (
 	}
 )
 
+// LivenessCheckController updates the availability of the liveness probe of the CA instance
+type LivenessCheckController struct {
+	rootCertFile       string
+	interval           time.Duration
+	grpcHostname       string
+	grpcPort           int
+	serviceIdentityOrg string
+	rsaKeySize         int
+	ca                 *ca.IstioCA
+	caCertTTL          time.Duration
+	livenessProbe      *probe.Probe
+	client             *grpcclient.CAGrpcClientImpl
+}
+
+func (c *LivenessCheckController) checkGrpcServer() error {
+	if c.grpcPort <= 0 {
+		return nil
+	}
+
+	// generates certificate and private key for test
+	opts := util.CertOptions{
+		Host:       livenessProbeClientIdentity,
+		RSAKeySize: 2048,
+	}
+
+	csrPEM, privPEM, err := util.GenCSR(opts)
+	if err != nil {
+		return err
+	}
+
+	certPEM, err := c.ca.Sign(csrPEM, c.interval, false)
+	if err != nil {
+		return err
+	}
+
+	testCert, err := ioutil.TempFile("/tmp", "cert")
```

Comment:
I think it's better that you make a temp dir and write the files in the temp dir: ioutil.TempDir(). This avoids potential conflicts with existing files in /tmp.

Reply:
TempFile has internal logic to avoid the conflict.  
I will create a temp directory to store certificates.